content/guides/accessibility/color-considerations.mdx

```diff
@@ -0,0 +1,332 @@
+
+#### Input text contrast
+
+Text portions of an input **must** meet a contrast ratio of `4.5:1` for "normal text", or `3:1` for "large text". This applies to both an input's visible label and its user-provided text content.
+
+<img src={InputText} alt="A wireframe illustration of a text input. Its label reads, 'Name.' It's user-provided text content reads, 'Mona Lisa.' A label pointing to the label reads, 'Communicates purpose.' A label pointing to the user provided content reads, 'Communicates user choice.'" />
+
+The majority of GitHub input text size is "normal text". Both "normal text" and "large text" are WCAG-specific terms. More information about this can be found on [the Key Terms subsection of SC 1.4.3: Contrast (Minimum)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html#dfn-large-scale).
+
+#### Input border contrast
+
+Borders communicate **the boundaries of an input**, and are considered non-text contrast. This includes inputs that allow for freeform user input such as text, and also pre-specified inputs such as checkboxes.
+
+Borders of an input **must** meet a contrast ratio of `3:1`. This is calculated by the border's computed color value compared to the computed background color the input is placed on.
+
+<img src={InputBorderContrast} alt="A wireframe illustration of a text input with a border. Its label reads, 'Email address.' A label pointing to the input's border reads, 'Communicates interactive area boundary.'" />
+
+The contrast of the input's border color compared to the input's background color **does not** need to meet a contrast ratio. The color of [the input's text against its background color](#input-text-contrast) still needs to be compliant, however.
+
+Inputs without visible borders must use their background color instead. In this case, the required contrast ratio is `3:1`, with the computed input background color and the color of the background of what the input is placed on.
+
+<img src={InputBorderContrast} alt="A wireframe illustration of a text input without a border. Its label reads, 'Phone number.' It's user-provided text content reads, '123-456-7890.' A label pointing to the input's background reads, 'Communicates interactive area boundary.'" />
+
+#### Input selected state contrast
+
+Selection state communicates **if a user has chosen an input option**, and is considered non-text contrast. Examples of this are a checkbox's check or a radio's dot.
+
+<img src={InputSelectionState} alt="A wireframe illustration of a checkbox input with a checkmark indicating it has been placed in a selected state. A label pointing to the checkmark reads, 'Communicates user choice.' A label pointing to the checkbox's border reads, 'Communicates interactive area boundary.'" />
+
+Selected state of an input **must** meet a contrast ratio of `3:1`. This is calculated by the border's computed color value compared to the computed background color the input is placed on.
```

Comment:
This one's a bit confusing to me because we've been talking about checkboxes. In this case, we just want to make sure that the selected state is identifiable from the non-selected state. The border may or may not have a part in this for instance, if our checkboxes didn't turn blue and just put a check in the middle, that would still pass. Sorry to put a little wrench in this a small wrench in this 😅

Reply:
Resolved via a pairing session with @ichelsea